### PR TITLE
[Slider] Update color themer

### PR DESCRIPTION
**Pull request by jmdetloff**
Before:
screen shot 2018-04-16 at 4 47 37 pm
screen shot 2018-04-16 at 4 47 42 pm
screen shot 2018-04-16 at 4 47 48 pm

After:
screen shot 2018-04-16 at 4 46 34 pm
screen shot 2018-04-16 at 4 46 41 pm
screen shot 2018-04-16 at 4 46 49 pm

### CODE REVIEW

File: components/Slider/src/ColorThemer/MDCSliderColorThemer.m [hunk 1]
```diff
@@ -19,9 +19,45 @@
 
 static const CGFloat kSliderThemerLightAlpha = 0.26f;
 static const CGFloat kSliderThemerDarkAlpha = 0.3f;
+static const CGFloat kSliderM2BaselineDisabledFillAlpha = 0.32;
+static const CGFloat kSliderM2BaselineEnabledBackgroundAlpha = 0.24;
+static const CGFloat kSliderM2BaselineDisabledBackgroundAlpha = 0.12;
+static const CGFloat kSliderM2BaselineEnabledTicksAlpha = 0.54;
+static const CGFloat kSliderM2BaselineDisabledTicksAlpha = 0.12;
```
Comment (jverkoey) [inline components/Slider/src/ColorThemer/MDCSliderColorThemer.m]: Remove "SliderM2Baseline" prefix on all of these constants. We'll eventually remove the kSliderThemerLightAlpha and kSliderThemerDarkAlpha as well.

**Review by jverkoey — APPROVED** [review]
LGTM pending trimming of const names.

**Comment by jverkoey** [comment]
```
ERROR: /Volumes/BuildData/tmpfs/src/github/repo/components/Slider/BUILD:35:1: C++ compilation of rule '//components/Slider:ColorThemer' failed (Exit 1)
components/Slider/src/ColorThemer/MDCSliderColorThemer.m:22:57: error: implicit conversion loses floating-point precision: 'double' to 'CGFloat' (aka 'float') [-Werror,-Wconversion]
static const CGFloat kSliderBaselineDisabledFillAlpha = 0.32;
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~   ^~~~
```

**Comment by jverkoey** [comment]
Is this PR blocked?

File: components/Slider/src/ColorThemer/MDCSliderColorThemer.m [hunk 2]
```diff
@@ -19,9 +19,45 @@
...
+static const CGFloat kSliderBaselineEnabledBackgroundAlpha = 0.24;
+static const CGFloat kSliderBaselineDisabledBackgroundAlpha = 0.12;
+static const CGFloat kSliderBaselineEnabledTicksAlpha = 0.54;
+static const CGFloat kSliderBaselineDisabledTicksAlpha = 0.12;
 
 @implementation MDCSliderColorThemer
 
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+                        toSlider:(nonnull MDCSlider *)slider {
+  slider.statefulAPIEnabled = YES;
```
Comment (romoore) [inline components/Slider/src/ColorThemer/MDCSliderColorThemer.m]: I don't think the color themer should be switching the Slider to the new API. I think the themer should instead check whether the new API is on or off, and then apply the scheme using the appropriate API.  The color/disableColor API is not yet deprecated and so should be supported.

**Review by romoore — DISMISSED** [review]
Please do not have a color themer flipping API implementation bits in the Slider.

File: components/Slider/src/ColorThemer/MDCSliderColorThemer.m [hunk 3]
```diff
@@ -19,9 +19,45 @@
 
 static const CGFloat kSliderThemerLightAlpha = 0.26f;
 static const CGFloat kSliderThemerDarkAlpha = 0.3f;
+static const CGFloat kSliderBaselineDisabledFillAlpha = 0.32;
```
Comment (romoore) [inline components/Slider/src/ColorThemer/MDCSliderColorThemer.m]: Please cast these to CGFloat.

File: components/Slider/examples/SliderCollectionViewController.m [hunk 4]
```diff
@@ -100,30 +102,33 @@ - (void)applyModel:(MDCSliderModel *)model {
...
   _slider.thumbHollowAtStart = model.hollowCircle;
   _slider.enabled = model.enabled;
   _slider.statefulAPIEnabled = YES;
+  
   // Don't apply a `nil` color, use the default
   if (model.sliderColor) {
     [_slider setTrackFillColor:model.sliderColor forState:UIControlStateNormal];
     [_slider setThumbColor:model.sliderColor forState:UIControlStateNormal];
     _slider.inkColor = model.sliderColor;
   }
```
Comment (romoore) [inline components/Slider/examples/SliderCollectionViewController.m:121]: You will need to explicitly set a default or the reused cells will start "inheriting" the green colors.  This was a bug I found while working on the APIs, but seem to have forgotten to commit the changes.

Comment (romoore) [reply components/Slider/examples/SliderCollectionViewController.m:121]: Oh, never mind. the "applyScheme" above should take care of that. Nice!

**Comment by romoore** [comment]
Why was this merged? This is modifying the slider API flag.

**Comment by romoore** [comment]
Ah, sorry. I misread and conflated the themer/test code. LGTM!

**Pull request merged**